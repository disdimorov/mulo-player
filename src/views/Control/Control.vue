<template>
  <button
    :title="title"
    class="control"
    type="button"
    @click.prevent="onClick">
    <img
      v-show="!isActive"
      :src="require(`@/assets/${iconName}.svg`)"
      :alt="alt"
      width="20"
      height="20">

    <img
      v-if="activeIconName"
      v-show="isActive"
      :src="require(`@/assets/${activeIconName}.svg`)"
      :alt="activeIconAlt"
      width="20"
      height="20">

    <slot/>
  </button>
</template>

<script>
export default {
  name: 'Control',

  props: {
    title: {
      type: String,
      required: true,
    },

    iconName: {
      type: String,
      required: true,
    },

    activeIconName: {
      type: String,
      default: '',
    },

    isActive: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    alt() {
      return `${this.iconName} icon`;
    },

    activeIconAlt() {
      return `${this.activeIconName} icon`;
    },
  },

  methods: {
    onClick() {
      this.$emit('click');
    },
  },
};
</script>

<style scoped lang="less">
// TODO color lib
@player-bg: #fff;
@player-border-color: darken(@player-bg, 12%);
@player-link-color: darken(@player-bg, 75%);

.control {
  display: block;
  padding: 16px;
  border-right: 1px solid @player-border-color;
  line-height: 0;
  color: @player-link-color;

  &:last-child {
    border-right: none;
  }
}
</style>
