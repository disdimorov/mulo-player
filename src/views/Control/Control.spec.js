import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import Control from './Control.vue';

const localVue = createLocalVue();

describe('Component Control', () => {
  const defaultProps = { title: 'Stop', iconName: 'stop' };
  const createWrapper = ({ isWithoutStubs = false, propsData = {}, ...options } = {}) => {
    const renderer = isWithoutStubs ? mount : shallowMount;
    const wrapper = renderer(Control, {
      localVue,
      propsData: Object.assign({}, defaultProps, propsData),
      ...options,
    });

    return { wrapper };
  };

  describe('snapshots', () => {
    it('should match snapshot with single icon', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should match snapshot with two icon and showing inactive icon', () => {
      const { wrapper } = createWrapper({
        isWithoutStubs: true,
        propsData: { activeIconName: 'play', isActive: true },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should match snapshot with single icon', () => {
      const { wrapper } = createWrapper();

      wrapper.trigger('click');
      expect(wrapper.emitted().click).toEqual([[]]);
    });
  });
});
