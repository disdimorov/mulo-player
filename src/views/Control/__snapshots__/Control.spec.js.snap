// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Control snapshots should match snapshot with single icon 1`] = `
<button title="Stop" type="button" class="control"><img src="" alt="stop icon" width="20" height="20">
  <!----> </button>
`;

exports[`Component Control snapshots should match snapshot with two icon and showing inactive icon 1`] = `<button title="Stop" type="button" class="control"><img src="" alt="stop icon" width="20" height="20" style="display: none;"> <img src="" alt="play icon" width="20" height="20"> </button>`;
